activetable.js

```diff
@@ -0,0 +1,117 @@
+var ActiveTable = (function () {
+    var  NUMERIC, STRING, check_response;
+
+    // Input type values must match the type values in the Python code.
+    NUMERIC = 1;
+    STRING = 2;
+
+    // Correctness checks for numeric and string inputs.  These functions are
+    // reimplementations of the corresponding functions in the Python code.
+    check_response = {};
+    check_response[NUMERIC] = function(self, student_response) {
+        var r = parseFloat(student_response);
+        if (isNaN(r)) return false;
+        return Math.abs(r - self.answer) <= self.abs_tolerance;
+    };
+    check_response[STRING] = function(self, student_response) {
+        return student_response == self.answer;
+    };
+
+    placeholder = {}
```

Comment:
@smarnach nit: not in `var` declaration at the top - essentially a global variable. Was it intentional?